api-objects/Appointment.ts

```diff
@@ -34,4 +39,46 @@ export class Appointment {
     public ths_sectionc: number;
     public ths_sectionc_example: string;
     public ts_type: string;
+
+    public createTfiMap() {
+        return new Map<string, number>([['overallTFI', this.tfi_overallscore], ['intrusive', this.tfi_i], ['sense', this.tfi_sc], ['cognitive', this.tfi_c], ['sleep', this.tfi_si], ['auditory', this.tfi_a], ['relaxation', this.tfi_r], ['quality', this.tfi_q], ['emotional', this.tfi_e]]);
+    }
+
+    public createThsScoreMap() {
+        return new Map<string, number>([['thsA', this.ths_sectiona], ['thsB', this.ths_sectionb], ['thsC', this.ths_sectionb]]);
+    }
+    public createThsTextVar() {
+        return new Map<string, string>([['thsCtxt', this.getthsCtxt()], ['thsCex', this.ths_sectionc_example]]);
+    }
+
+    public createTestSeverityMap() {
+        return new Map<string, string>([['audiogramType', this.audiogramtype], ['leftHighSev', this.leftear_highf_severity], ['leftLowSev', this.leftear_lowf_severity], ['rightHighSev', this.rightear_highf_severity], ['rightLowSev', this.rightear_lowf_severity], ['otoscopyType', this.otoscopytype], ['tympanometryType', this.tympanometrytype]]);
+    }
+
+    public createTestConfigurationMap() {
+        return new Map<string, string>([['leftHighConfig', this.leftear_highf_configuration.replace(/;/g, ', ')], ['leftLowConfig', this.leftear_lowf_configuration.replace(/;/g, ', ')], ['rightHighConfig', this.rightear_highf_configuration.replace(/;/g, ', ')], ['rightLowConfig', this.rightear_lowf_configuration.replace(/;/g, ', ')]]);
+    }
+    answerStrings = new ThsAnswerStrings();
+
+    public getthsCtxt() {
+        // let answerStrings = new ThsAnswerStrings();
+        switch (this.ths_sectionc) {
+            case 1:
+                return this.answerStrings.NO;
+            case 2:
+                return this.answerStrings.SMALL_YES;
+            case 3:
+                return this.answerStrings.MODERATE_YES;
+            case 4:
+                return this.answerStrings.BIG_YES;
+            case 5:
+                return this.answerStrings.VERY_BIG_YES;
+        }
+    }
+
+    constructor(clone?: Appointment) {
+        AppointmentVariables.forEach((value) => {
+            this[value] = clone[value];
```

Comment:
`clone` is an optional argument, but this seems to fail if `clone` is undefined.

Reply:
🤦‍♂ I completely forgot to add the if statement back.